state/user.go

```diff
@@ -129,6 +142,66 @@ func (st *State) addUser(name, displayName, password, creator string, secretKey
 	return user, nil
 }
 
+// updateExistingUser manipulates the values of an existing user in the db.
+// This is particularly useful when reusing existing users that were previously
+// deleted.
+func (st *State) updateExistingUser(u *User, displayName, password, creator string, secretKey []byte) (*User, error) {
```

Comment:
@juanmanuel-tirado 
The `TestAddRemovedUser` failed in 3.0:
`failed to create user: transaction aborted`
There might be some assertion failed in this method.
I don't have enough context for this change.
It'd be great if you could fix it by pushing to this PR directly.